File: ido_sdk_flutter/ido_sdk/protocol_sdk/protocol_channel/ios/Classes/Model/IDOMusicControlParamModel.swift
//
//  IDOMusicControlParamModel.swift
//  protocol_channel
//
//  Created by hc on 2023/10/17.
//

import Foundation

@objcMembers
public class IDOMusicControlParamModel: NSObject, IDOBaseModel {
    /// Status: 0: Invalid 1: Play 2: Pause 3: Stop
    public var status: Int
    /// Current play time Unit:second
    public var curTimeSecond: Int
    /// Total play time Unit:second
    public var totalTimeSecond: Int
    /// Music name (maximum 63 bytes)
    public var musicName: String
    /// Singer name (maximum 63 bytes)
    /// This value is not applicable if v3MusicControl02AddSingerName is not enabled on the firmware
    public var singerName: String

    enum CodingKeys: String, CodingKey {
        case status
        case curTimeSecond = "cur_time_second"
        case totalTimeSecond = "total_time_second"
        case musicName = "music_name"
        case singerName = "singer_name"
    }

    public init(status: Int, curTimeSecond: Int, totalTimeSecond: Int, musicName: String, singerName: String) {
        self.status = status
        self.curTimeSecond = curTimeSecond
        self.totalTimeSecond = totalTimeSecond
        self.musicName = musicName
        self.singerName = singerName
    }
    
    public func toJsonString() -> String? {
        if let jsonData = try? JSONEncoder().encode(self) {
            let jsonString = String(data: jsonData, encoding: .utf8)
            return jsonString
        }
        return nil
    }
}

File: ido_sdk_flutter/ido_sdk/protocol_sdk/protocol_channel/android/src/main/kotlin/com/idosmart/model/IDOMusicControlParamModel.kt
//
//  IDOMusicControlParamModel.kt
//  protocol_channel
//
//  Created by hedongyang on 2023/10/30.
//

package com.idosmart.model

import com.google.gson.Gson
import com.google.gson.GsonBuilder
import com.google.gson.annotations.SerializedName

open class IDOMusicControlParamModel(
    status: Int,
    curTimeSecond: Int,
    totalTimeSecond: Int,
    musicName: String,
    singerName: String
) : IDOBaseModel {

    /// Status: 0: Invalid 1: Play 2: Pause 3: Stop
    @SerializedName("status")
    var status: Int = status

    /// Current play time Unit:second
    @SerializedName("cur_time_second")
    var curTimeSecond: Int = curTimeSecond

    /// Total play time Unit:second
    @SerializedName("total_time_second")
    var totalTimeSecond: Int = totalTimeSecond

    /// Music name (maximum 63 bytes)
    @SerializedName("music_name")
    var musicName: String = musicName

    /// Singer name (maximum 63 bytes)
    /// This value is not applicable if v3MusicControl02AddSingerName is not enabled on the firmware
    @SerializedName("singer_name")
    var singerName: String = singerName


    

    override fun toJsonString(): String {
        return GsonBuilder().create().toJson(this).toString()
    }
}
    